src/atomicui/atoms/NLLoader/NLLoader.tsx

```diff
@@ -0,0 +1,48 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: MIT-0 */
+
+import React, { useEffect, useState } from "react";
+
+import { Loader as AmplifyLoader, Flex } from "@aws-amplify/ui-react";
+import { useTranslation } from "react-i18next";
+
+interface NLLoaderProps {
+	nlLoadText: string[];
+}
+const NLLoader: React.FC<NLLoaderProps> = ({ nlLoadText }) => {
+	const [loaderIdx, setloaderIdx] = useState(0);
+	const { t } = useTranslation();
+
+	useEffect(() => {
+		let timeout: ReturnType<typeof setTimeout>;
+		if (loaderIdx < nlLoadText.length - 1) {
```

Comment:
I would imagine we would go with typing animation and a random selection of text message.